net/tailscale/files/tailscale.init

```diff
@@ -4,44 +4,266 @@
-  local port
-  local std_err std_out
+PROG=/usr/sbin/tailscale
+PROGD=/usr/sbin/tailscaled
+CONFIG_PATH=/var/lib/tailscale
+
+service_triggers() {
+  procd_add_reload_trigger "tailscale"
+  procd_add_interface_trigger "interface.*.up" wan /etc/init.d/tailscale reload
+}
 
-  config_load tailscale
-  config_get_bool std_out "settings" log_stdout 1
-  config_get_bool std_err "settings" log_stderr 1
-  config_get port "settings" port 41641
-  config_get state_file "settings" state_file /etc/tailscale/tailscaled.state
-  config_get fw_mode "settings" fw_mode nftables
+section_enabled() {
+  config_get_bool enabled "$1" 'enabled' 0
+  [ $enabled -gt 0 ]
+}
 
-  /usr/sbin/tailscaled --cleanup
+custom_instance() {
+  local cfg="$1"
+  local accept_routes hostname accept_dns advertise_exit_node exit_node advertise_routes site_to_site subnet_routes flags login_server authkey std_out std_err
+
+  config_get_bool accept_routes $cfg 'accept_routes'
+  config_get hostname $cfg 'hostname'
+  config_get_bool accept_dns $cfg 'accept_dns'
+  config_get_bool advertise_exit_node $cfg 'advertise_exit_node'
+  config_get exit_node $cfg 'exit_node'
+  config_get advertise_routes $cfg 'advertise_routes'
+  config_get_bool site_to_site $cfg 'site_to_site'
+  config_get flags $cfg 'flags'
+  config_get login_server $cfg 'login_server'
+  config_get authkey $cfg 'authkey'
+  config_get_bool std_out $cfg 'log_stdout'
+  config_get_bool std_err $cfg 'log_stderr'
 
   procd_open_instance
-  procd_set_param command /usr/sbin/tailscaled
+  procd_set_param command $PROG up --reset
 
-  # Starting with v1.48.1 ENV variable is required to enable use of iptables / nftables.
-  # Use nftables by default - can be changed to 'iptables' in tailscale config
-  procd_set_param env TS_DEBUG_FIREWALL_MODE="$fw_mode"
+  [ "$accept_routes" = "1" ] && procd_append_param command --accept-routes=true
+  [ -n "$hostname" ] && procd_append_param command --hostname="$hostname"
+  [ "$accept_dns" = "0" ] && procd_append_param command --accept-dns=false
+  [ "$advertise_exit_node" = "1" ] && procd_append_param command --advertise-exit-node
+  [ -n "$exit_node" ] && {
+    procd_append_param command --exit-node="$exit_node"
+    procd_append_param command --exit-node-allow-lan-access=true
+  }
+  [ -n "$advertise_routes" ] && procd_append_param command --advertise-routes="$(echo $advertise_routes | tr ' ' ',')"
+  [ "$site_to_site" = "1" ] && procd_append_param command --snat-subnet-routes=false
+  [ -n "$flags" ] && procd_append_param command $flags
+  [ -n "$login_server" ] && procd_append_param command --login-server="$login_server"
+  [ -n "$authkey" ] && procd_append_param command --authkey="$authkey"
+
+  procd_set_param stdout "$std_out"
+  procd_set_param stderr "$std_err"
+  procd_close_instance
+  (
+    # Check if the wait PID file exists, if so, return
+    [ -f "/var/run/tailscale.wait.pid" ] && return
+    # Create the wait PID file
+    touch /var/run/tailscale.wait.pid
```

Comment:
I think this whole section needs more error handling. What happens if any part of this fails? It looks like it can't be re-entered within the same reboot cycle, because the _lock_ file will still be there.

You should probably use flock and add a cleanup trap.

Reply:
The error handling part is still relevant.